src/client/GameSave.cpp

```diff
@@ -749,62 +735,62 @@ void GameSave::readOPS(const std::vector<char> &data)
 	//Read wall and fan data
 	if(wallData)
 	{
+		// TODO: use PlaneAdapter<std::span<unsigned char>> once we're C++20
+		auto wallDataPlane = PlaneAdapter<const std::basic_string_view<unsigned char>>(blockS, std::in_place, wallData, blockS.X * blockS.Y);
 		unsigned int j = 0;
-		if (blockW * blockH > wallDataLen)
+		if (blockS.X * blockS.Y > int(wallDataLen))
 			throw ParseException(ParseException::Corrupt, "Not enough wall data");
-		for (unsigned int x = 0; x < blockW; x++)
-		{
-			for (unsigned int y = 0; y < blockH; y++)
-			{
-				if (wallData[y*blockW+x])
-					blockMap[blockY+y][blockX+x] = wallData[y*blockW+x];
-
-				if (blockMap[y][x]==O_WL_WALLELEC)
-					blockMap[y][x]=WL_WALLELEC;
-				if (blockMap[y][x]==O_WL_EWALL)
-					blockMap[y][x]=WL_EWALL;
-				if (blockMap[y][x]==O_WL_DETECT)
-					blockMap[y][x]=WL_DETECT;
-				if (blockMap[y][x]==O_WL_STREAM)
-					blockMap[y][x]=WL_STREAM;
-				if (blockMap[y][x]==O_WL_FAN||blockMap[y][x]==O_WL_FANHELPER)
-					blockMap[y][x]=WL_FAN;
-				if (blockMap[y][x]==O_WL_ALLOWLIQUID)
-					blockMap[y][x]=WL_ALLOWLIQUID;
-				if (blockMap[y][x]==O_WL_DESTROYALL)
-					blockMap[y][x]=WL_DESTROYALL;
-				if (blockMap[y][x]==O_WL_ERASE)
-					blockMap[y][x]=WL_ERASE;
-				if (blockMap[y][x]==O_WL_WALL)
-					blockMap[y][x]=WL_WALL;
-				if (blockMap[y][x]==O_WL_ALLOWAIR)
-					blockMap[y][x]=WL_ALLOWAIR;
-				if (blockMap[y][x]==O_WL_ALLOWSOLID)
-					blockMap[y][x]=WL_ALLOWPOWDER;
-				if (blockMap[y][x]==O_WL_ALLOWALLELEC)
-					blockMap[y][x]=WL_ALLOWALLELEC;
-				if (blockMap[y][x]==O_WL_EHOLE)
-					blockMap[y][x]=WL_EHOLE;
-				if (blockMap[y][x]==O_WL_ALLOWGAS)
-					blockMap[y][x]=WL_ALLOWGAS;
-				if (blockMap[y][x]==O_WL_GRAV)
-					blockMap[y][x]=WL_GRAV;
-				if (blockMap[y][x]==O_WL_ALLOWENERGY)
-					blockMap[y][x]=WL_ALLOWENERGY;
-
-				if (blockMap[y][x] == WL_FAN && fanData)
+		for (auto bpos : blockS.OriginRect().Range<LEFT_TO_RIGHT, TOP_TO_BOTTOM>())
+		{
+			auto &bm = blockMap[blockP + bpos];
+			if (wallDataPlane[bpos])
+				bm = wallDataPlane[bpos];
```

Comment:
Why not
```cpp
auto bm = wallDataPlane[bpos];
switch (bm) { ... }
blockMap[blockP + bpos] = bm;
```

Reply:
No reason, I like my references. Will go with yours.